src/nectarchain/calibration/NectarGain/SPEfit/NectarGainSPE.py

```diff
@@ -0,0 +1,200 @@
+            ax.set_xlabel("Charge (ADC)", size=15)
+            ax.set_ylabel("Events", size=15)
+            ax.legend(fontsize=15)
+            os.makedirs(f"{os.environ.get('NECTARCHAIN_LOG')}/{os.getpid()}/figures/",exist_ok=True)
+            fig.savefig(f"{os.environ.get('NECTARCHAIN_LOG')}/{os.getpid()}/figures/initialization_mean_pixel{extension}_{os.getpid()}.pdf")
+            fig.clf()
+            plt.close(fig)
+            del fig,ax
+        return coeff_mean,var_matrix
+
+    @staticmethod
+    def _get_pedestal_gaussian_fit(charge_in, histo_in ,extension = "") :
+        #x = np.linspace(nectargain.charge[pixel].min(),nectargain.charge[pixel].max(),int(nectargain.charge[pixel].max()-nectargain.charge[pixel].min()))
+        #interp = interp1d(nectargain.charge[pixel],nectargain.histo[pixel])
+        charge = charge_in.data[~histo_in.mask]
+        histo = histo_in.data[~histo_in.mask]
+
+        windows_lenght = 80
+        order = 2
+        histo_smoothed = savgol_filter(histo, windows_lenght, order)
+
+        peaks = find_peaks(histo_smoothed,10)
+        peak_max = np.argmax(histo_smoothed[peaks[0]])
+        peak_pos,peak_value = charge[peaks[0][peak_max]], histo[peaks[0][peak_max]]
+
+        coeff, var_matrix = curve_fit(weight_gaussian, charge[:peaks[0][peak_max]], histo_smoothed[:peaks[0][peak_max]],p0 = [peak_value,peak_pos,1])
+
+        if log.getEffectiveLevel() == logging.DEBUG :
+            log.debug('plotting figures with prefit parameters computation') 
+            fig,ax = plt.subplots(1,1,figsize = (8,8))
+            ax.errorbar(charge,histo,np.sqrt(histo),zorder=0,fmt=".",label = "data")
+            ax.plot(charge,histo_smoothed,label = f'smoothed data with savgol filter (windows lenght : {windows_lenght}, order : {order})')
+            ax.plot(charge,weight_gaussian(charge,coeff[0],coeff[1],coeff[2]),label = 'gaussian fit of the pedestal, left tail only')
+            ax.vlines(coeff[1],0,peak_value,label = f'pedestal initial value = {coeff[1]:.0f}',color = 'red')
+            ax.add_patch(Rectangle((coeff[1]-coeff[2], 0), 2 * coeff[2], peak_value,fc=to_rgba('red', 0.5)))
+            ax.set_xlabel("Charge (ADC)", size=15)
+            ax.set_ylabel("Events", size=15)
+            ax.legend(fontsize=15)
+            os.makedirs(f"{os.environ.get('NECTARCHAIN_LOG')}/{os.getpid()}/figures/",exist_ok=True)
+            fig.savefig(f"{os.environ.get('NECTARCHAIN_LOG')}/{os.getpid()}/figures/initialization_pedestal_pixel{extension}_{os.getpid()}.pdf")
+            fig.clf()
+            plt.close(fig)
+            del fig,ax
+        return coeff,var_matrix
+
+    @abstractmethod
+    def create_output_table(self) : pass
+
+    @abstractmethod
+    def save(self,path,**kwargs) : pass
+    
+    @abstractmethod
+    def run(self,pixel : int = None,**kwargs): pass
+    @abstractmethod
+    def _run_obs(self,pixel : int,**kwargs) : pass
+    @abstractclassmethod
```

Comment:
Same question about PyCharm warning me that: `abc.abstractclassmethod` is deprecated since Python 3.3. Use `classmethod` with `abc.abstractmethod` instead.

Reply:
I have changed following the method documented here https://docs.python.org/3/library/abc.html